stdlib/public/core/FixedPoint.swift.gyb

```diff
@@ -429,44 +426,62 @@ extension ${Self} {
 extension ${Self} : SignedNumber {}
 % end
 
-// Construction from other integer types
-@_transparent
-extension ${Self} {
+
 % for src_ty in all_integer_types(word_bits):
 %    srcBits = src_ty.bits
 %    srcSigned = src_ty.is_signed
 %    Src = src_ty.stdlib_name
 %    (srcSign, srcExt) = ('s', 'sext') if srcSigned else ('u', 'zext')
-%    if Self != Src:
-  public init(_ v: ${Src}) {
+%    conversionFunction = "_convert" + Src + "to" + Self
+
+@_transparent
+private func ${conversionFunction}(_ value: ${Src}) -> (value: Builtin.${BuiltinName}, error: Bool) {
```

Comment:
Please don't use `private` in the standard library (there are compiler issues), use `internal` instead.